### PR TITLE
Update data-soft-delete.md with MV soft delete capability

### PR DESCRIPTION
**Pull request by tomersh26**
Update the document with the recently new capability of MV soft delete

### CODE REVIEW

**Comment by Jak-MS** [comment]
@shsagir 
- Can you review this PR?
- IMPORTANT: When this content is ready to merge, you must add `#sign-off` in a comment or the approval may get overlooked.


#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

File: data-explorer/kusto/concepts/data-soft-delete.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ The soft delete process is performed using the following steps:
 
 * Soft delete is only available on clusters running Engine V3.
 
-* Soft delete is only supported for native tables and isn't supported for external tables or materialized views.
+* Soft delete is only supported for native tables and materialized views. It isn't supported for external tables.
```
Comment (yifatsor) [inline data-explorer/kusto/concepts/data-soft-delete.md:34]: I think you can remove the "only" in this sentence, now that it supports MVs as well

File: data-explorer/kusto/concepts/data-soft-delete.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ The soft delete process is performed using the following steps:
 * Soft delete is executed against your engine endpoint: `https://[YourClusterName].[region].kusto.windows.net`. The command requires [database admin](../management/access-control/role-based-access-control.md) permissions on the relevant database.
 
 * Soft delete can affect materialized views based on a source table in which records are deleted. This can happen because every [materialization cycle](../management/materialized-views/materialized-view-overview.md#how-materialized-views-work) adds newly ingested data to the materialized part from the previous cycle. Therefore, if the command deletes newly ingested records before a new cycle begins, those records won't be added to the materialized view. Otherwise, deleting records won't affect the materialized view.
```
Comment (yifatsor) [inline data-explorer/kusto/concepts/data-soft-delete.md:44]: I would add a `##` section for `Soft delete for table`, just like you added for materialized views. Add this bullet about how soft delete affects MVs under that section (it's only relevant when deleting from the source table of an MV, not when deleting from the MV itself). 
Additionally, you can remove the note below ("soft delete is also supported for materialized views")

File: data-explorer/kusto/concepts/data-soft-delete.md [hunk 3]
```diff
@@ -110,3 +111,16 @@ To delete all the records that contain data of a given user:
...
+Soft deleted can be executed also on materialized views, where the same concepts explained above are applied similarly for materialized views.
+
+### Syntax
+
+`.delete` [`async`] `materialized-view` *MaterializedViewName* `records` [`with (`propertyName `=` propertyValue [`,` ...]`)`] `<|` *Predicate*
+
+> [!NOTE]
+> The same restrictions on the *Predicate* mentioned above apply here as well.
+> Soft delete is expected to fail in case a materialization of the view is being executed concurrently (specifically, in case there is an intersection between the extents those two commands are executing on).
+> Usage of function 'materialized_view' is not allowed in *Predicate*.
```
Comment (yifatsor) [inline data-explorer/kusto/concepts/data-soft-delete.md:126]: `materialized_view()` should link to the page of the function and should use `` (not '').

File: data-explorer/kusto/concepts/data-soft-delete.md [hunk 4]
```diff
@@ -110,3 +111,16 @@ To delete all the records that contain data of a given user:
 ### Output
 
 The output of the command contains information about which extents were replaced.
+
+## Soft delete for materialized views
+
+Soft deleted can be executed also on materialized views, where the same concepts explained above are applied similarly for materialized views.
+
+### Syntax
```
Comment (yifatsor) [inline data-explorer/kusto/concepts/data-soft-delete.md:119]: please add an example for MV as well (just like in the tables section)

Comment (yifatsor) [reply data-explorer/kusto/concepts/data-soft-delete.md:119]: and the Parameters section as well (like in the tables section)

File: data-explorer/kusto/concepts/data-soft-delete.md [hunk 5]
```diff
@@ -110,3 +111,16 @@ To delete all the records that contain data of a given user:
...
+
+`.delete` [`async`] `materialized-view` *MaterializedViewName* `records` [`with (`propertyName `=` propertyValue [`,` ...]`)`] `<|` *Predicate*
+
+> [!NOTE]
+> The same restrictions on the *Predicate* mentioned above apply here as well.
+> Soft delete is expected to fail in case a materialization of the view is being executed concurrently (specifically, in case there is an intersection between the extents those two commands are executing on).
```
Comment (yifatsor) [inline data-explorer/kusto/concepts/data-soft-delete.md:125]: I would rephrase to - "Soft delete might fail in case of conflicts with the [materialization process](link to https://learn.microsoft.com/en-us/azure/data-explorer/kusto/management/materialized-views/materialized-view-overview#how-materialized-views-work) running in the background. Retrying the operation can help in this case. To avoid conflicts, you can [disable the materialized view](link to https://learn.microsoft.com/en-us/azure/data-explorer/kusto/management/materialized-views/materialized-view-enable-disable) before executing soft delete, and re-enable it when the operation completes."

**Comment by yifatsor** [comment]
+ @YaelSchuster  can you please review as well?

File: data-explorer/kusto/concepts/data-soft-delete.md (hunk 3 above)
Comment (tomersh26) [reply data-explorer/kusto/concepts/data-soft-delete.md:126]: If I'm already adding a link, I think it's ok to remove the apostrophe

File: data-explorer/kusto/concepts/data-soft-delete.md [hunk 6]
```diff
@@ -110,3 +110,40 @@ To delete all the records that contain data of a given user:
...
+|--|--|--|--|
+|`async`|string||If specified, indicates that the command runs in asynchronous mode.|
+|*MaterializedViewName*|string|&check;|The name of the materialized view from which to delete records.|
+|*Predicate*|string|&check;|The predicate that returns records to delete. Specified as a query.|
+
+> [!NOTE]
+> The same restrictions on the *Predicate* mentioned for table apply here as well.
+> Soft delete might fail in case of conflicts with the [materialization process](../management/materialized-views/materialized-view-overview.md#how-materialized-views-work) running in the background. Retrying the operation can help in this case. To avoid conflicts, you can [disable the materialized view](../management/materialized-views/materialized-view-enable-disable.md) before executing soft delete, and re-enable it when the operation completes.
+> Usage of function [materialized_view()](../query/materialized-view-function.md) is not allowed in *Predicate*.
+>
```
Comment (YaelSchuster) [inline data-explorer/kusto/concepts/data-soft-delete.md:134]:
```suggestion

```

**Comment by YaelSchuster** [comment]
@yifatsor @tomersh26 I'll pull this PR into the private repo to check Acrolinx, make edits, and check staging. FYI we are on a content freeze until May 23, evening ILDC time.

**Comment by YaelSchuster** [comment]
Recreated in internal docs repo with edits https://github.com/MicrosoftDocs/dataexplorer-docs-pr/pull/4459

**Comment by YaelSchuster** [comment]
#please-close
